ThinDownloadManager/src/main/java/com/thin/downloadmanager/DownloadRequestQueue.java

```diff
@@ -230,9 +233,13 @@ void pauseAll() {
 	private void checkResumableDownloadEnabled(int downloadId) {
 		synchronized (mCurrentRequests) {
 			for (DownloadRequest request : mCurrentRequests) {
-					if (downloadId == -1 ||
-							(request.getDownloadId() == downloadId && !request.isResumable())) {
-						throw new IllegalStateException("You cannot pause the download, unless your request enabled Resume feature in DownloadRequest.");
+				if (downloadId == -1 && !request.isResumable()) {
+					Log.e("ThinDownloadManager",
```

Comment:
If we haven't enabled the resumable feature for this request hence canceling the request.
(we will not throw exception for all requests)